barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_integration.test.cpp

```diff
@@ -66,16 +62,65 @@ class AcirIntegrationTest : public ::testing::Test {
 
         return verifier.verify_proof(proof);
     }
-};
 
-class AcirIntegrationSingleTest : public AcirIntegrationTest, public testing::WithParamInterface<std::string> {
+    template <class Flavor> bool prove_and_verify_honk(std::shared_ptr<ProverInstance_<Flavor>> prover_instance)
+    {
+        using Prover = UltraProver_<Flavor>;
+        using Verifier = UltraVerifier_<Flavor>;
+        using VerificationKey = Flavor::VerificationKey;
+
+        Prover prover{ prover_instance };
+        auto proof = prover.construct_proof();
+
+        // Verify Honk proof
+        auto verification_key = std::make_shared<VerificationKey>(prover.instance->proving_key);
+        Verifier verifier{ verification_key };
+
+        return verifier.verify_proof(proof);
+    }
+
+    void add_some_simple_RAM_gates(auto& circuit)
+    {
+        std::array<uint32_t, 3> ram_values{ circuit.add_variable(5),
+                                            circuit.add_variable(10),
+                                            circuit.add_variable(20) };
+
+        size_t ram_id = circuit.create_RAM_array(3);
+
+        for (size_t i = 0; i < 3; ++i) {
+            circuit.init_RAM_element(ram_id, i, ram_values[i]);
+        }
+
+        auto val_idx_1 = circuit.read_RAM_array(ram_id, circuit.add_variable(1));
+        auto val_idx_2 = circuit.read_RAM_array(ram_id, circuit.add_variable(2));
+        auto val_idx_3 = circuit.read_RAM_array(ram_id, circuit.add_variable(0));
+
+        circuit.create_big_add_gate({
+            val_idx_1,
+            val_idx_2,
+            val_idx_3,
+            circuit.zero_idx,
+            1,
+            1,
+            1,
+            0,
+            -35,
+        });
+    }
+
   protected:
     static void SetUpTestSuite() { srs::init_crs_factory("../srs_db/ignition"); }
 };
 
+class AcirIntegrationSingleTest : public AcirIntegrationTest, public testing::WithParamInterface<std::string> {};
```

Comment:
weird diff here. This class already existed I just moved the bn254 init to the base class